cpp/benchmarks/copying/contiguous_split_benchmark.cu

```diff
@@ -106,31 +117,46 @@ void BM_contiguous_split_strings(benchmark::State& state)
   cudf::size_type num_cols    = state.range(1);
   cudf::size_type num_splits  = state.range(2);
   bool include_validity       = state.range(3) == 0 ? false : true;
+  using string_pair           = thrust::pair<const char*, cudf::size_type>;
 
   const int64_t string_len = 8;
   std::vector<const char*> h_strings{
     "aaaaaaaa", "bbbbbbbb", "cccccccc", "dddddddd", "eeeeeeee", "ffffffff", "gggggggg", "hhhhhhhh"};
+  auto [chars, offsets] = cudf::test::detail::make_chars_and_offsets(
+    h_strings.begin(), h_strings.end(), thrust::make_constant_iterator(true));
+  auto d_chars = cudf::detail::make_device_uvector_sync(chars);
+  rmm::device_uvector<string_pair> d_strings(h_strings.size(), rmm::cuda_stream_default);
+  thrust::tabulate(thrust::device,
+                   d_strings.begin(),
+                   d_strings.end(),
+                   [data = d_chars.data()] __device__(size_t i) {
+                     return string_pair{data + i * string_len, string_len};
+                   });
```

Comment:
This seems like its more complicated than it should be. How about something like this?
```
auto d_strings = cudf::test::strings_column_wrapper({"aaaaaaaa", "bbbbbbbb", "cccccccc",  "dddddddd",
                                                     "eeeeeeee", "ffffffff", "gggggggg", "hhhhhhhh", ""},
                                                    {1,1,1,1, 1,1,1,1, 0}); // include one null element

// use the random_elements iterator created below to build a gather-map
auto d_elements = cudf::tests::fixed_width_column_wrapper<cudf::size_type>( random_elements,
                                                                            random_elements + num_rows );

// gather for strings is pretty fast now
auto d_table = cudf::gather( cudf::table({d_strings}), d_elements, cudf::out_of_bounds_policy::DONT_CHECK );
auto d_col = d_table.column(0); // I suppose this should be column-wrapper to fit here

```
This seems simpler and removes the need for string-pairs and `thrust::tabulate`.

Reply:
I like this idea. 
But result of gather is always `nullable` even if `has_nulls == false`, which we don't want in this benchmark.
Any alternatives for this?

Reply:
I think something like the following will work to remove the nullable:
```
auto d_col = d_table->release().first();
if ( !d_col->has_nulls() ) d_col->set_null_mask( rmm::device_buffer{}, 0 );
```